enterprise-suite/tests/smoke_prometheus

```diff
@@ -250,6 +252,20 @@ busy_wait app_data_via_service
 
 prom_has_data "$app_instances_via_service"
 
+
+monitor_exists() {
+  results=$( prom_query "model{name=\"$1\"}" | jq '.data.result | length' )
+  ! [[ results -eq 0 ]]
+}
+
+# kubernetes-cadvisor metrics should have an es_monitor_type label.
+curl -XPOST -H "Content-Type: application/json" -H 'Author-Name: Me' -H 'Author-Email: me@lightbend.com' -H 'Message: testing!' \
+    "$ES_MONITOR_API/monitors/es-test/es-monitor-type-test" \
+    --data '{"model":"threshold","parameters":{"comparator":"\u003e=","confidence":"1","description":"cpu on {{ $labels.instance }} over 99% busy","meta":{},"metric":"container_cpu_load_average_10s{es_workload=\"es-test\"}","monitorType":"es-test","name":"es-monitor-type-test","severity":"warning","summary":"cpu saturated","threshold":"99","window":"5m"}}'
+
+busy_wait monitor_exists es-monitor-type-test
+prom_has_data '{job="kubernetes-cadvisor",es_monitor_type="es-test"}'
```

Comment:
this is great!

This test says: something in container data should have a monitor_type label. That's good, but doesn't say what data must have that label, and doesn't cover the other "job" types.

We should now be able to have a very strong test against the requirement:

```busy wait prom_has_no_data  '{es_workload="es-test", es_monitor_type != "es-test"}'```

meaning... everything with this workload label should (eventually) have this monitor type.